### PR TITLE
feat: show error after submit. show message to the user

### LINKED ISSUE #97: show error message on invalid input
Better than status quo so I'll merge it but we need to also include a few things:

1. error message explaining why invalid
2. ideally no error when you're first typing it in, only show errors once you've tried to submit once

Also ideally we wouldn't even make the button disabled, people unfamiliar with computers tend to get confused with non-clickable button-looking elements.

Screen Shot 2021-02-10 at 15 17 48

_Originally posted by @osdiab in https://github.com/everydotorg/donate-button/issues/95#issuecomment-776473311_

### PR DESCRIPTION
**Pull request by martinbianchi**
closes #97 

I also added a validation for amounts greater than 1,000,000

### CODE REVIEW

File: packages/donate-button/src/components/Input/input.css [hunk 1]
```diff
@@ -17,7 +17,7 @@
 }
 
 .input--error {
-    border-color: red!important
+    border-color: orangered!important
```
Comment (osdiab) [inline packages/donate-button/src/components/Input/input.css:20]: Ideally styles wouldn't need !important, I'll add a lint rule banning that later, but it's fine for now given that a lot of the styles will likely be revamped when tackling removing style injector in #77

**Review by osdiab — APPROVED** [review]
Will merge once docs are built and committed

**Review by osdiab — CHANGES_REQUESTED** [review]
tried it out, still some UX issues - see here

https://app.usebubbles.com/q9tZwqXLSNVRLqMit4cyG1/comments-on-every-month-git-mbianchi-error-message-p22-verce

**Comment by martinbianchi** [comment]
@osdiab I saw the video, but seems that the changes weren't impacted. The same with #103 . Could be something wrong with the deployment? Or should I do something before push the changes?

**Comment by osdiab** [comment]
Oh yeah maybe the Vercel demo isn't working unless you build and commit? I'll check tomorrow if that's the case, would be better if it weren't.

**Comment by osdiab** [comment]
oh, i see - the vercel instance run by p22 (`every-month`) doesn't update but the one run by us at Every.org (`donate-button`) does. @jake-nz can you turn off the `every-month` vercel instance?

**Comment by osdiab** [comment]
I'll merge this but follow up issue, the error messages shouldn't cause the button to move.

**Comment by jake-nz** [comment]
> oh, i see - the vercel instance run by p22 (`every-month`) doesn't update but the one run by us at Every.org (`donate-button`) does. @jake-nz can you turn off the `every-month` vercel instance?

Done ✅

**Pull request merged**